pkg/graveler/ref/manager.go

```diff
@@ -142,6 +145,10 @@ func (m *Manager) getRepository(ctx context.Context, repositoryID graveler.Repos
 		}
 		return nil, err
 	}
+	repo := graveler.RepoFromProto(&data)
+	if repo.StorageID == config.SingleBlockstoreID {
+		repo.StorageID = graveler.StorageID(m.storageConfig.GetStorageByID(config.SingleBlockstoreID).ID()) // Will return the real actual ID
```

Comment:
Noting that on server start `storageConfig.GetStorageByID()` is checked for each repo, so it can't return null here.

Reply:
Fixed